### PR TITLE
It's time to #deleteDart1

### PR DESCRIPTION
**Pull request by cbracken**
Eliminates --preview-dart-2 flag from flutter_tools. All commands now run Dart 2 only.

### CODE REVIEW

File: packages/flutter_tools/gradle/flutter.gradle [hunk 1]
```diff
@@ -383,12 +376,10 @@ abstract class BaseFlutterTask extends DefaultTask {
         if (buildMode != 'debug') {
             // For AOT builds, include the gen_snapshot depfile.
             FileCollection depfiles = project.files("${intermediateDir}/snapshot.d")
-            if (previewDart2) {
-                // For Dart 2, also include the kernel compiler depfile, since
-                // kernel compile is the first stage of AOT build in this mode,
-                // and it includes all the Dart sources.
-                depfiles += project.files("${intermediateDir}/kernel_compile.d")
-            }
+            // For Dart 2, also include the kernel compiler depfile, since
```
Comment (gspencergoog) [inline packages/flutter_tools/gradle/flutter.gradle]: You can just say "Include the kernel...", since there is no Dart 1...

**Comment by cbracken** [comment]
Apologies for the giant patch. Each of the commits is *pretty* self-contained, which should make review a bit easier. 

Since there's really no case where it makes sense to partially force the tool to operate in Dart 2 mode, the plan is to squash and merge in one go once it's through review.

**Review by gspencergoog — APPROVED** [review]

**Comment by gspencergoog** [comment]
Luckily there are a lot of deletes.  Those are pretty easy to review. :-)

**Comment by gspencergoog** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter_tools/gradle/flutter.gradle (hunk 1 above)
Comment (cbracken) [reply packages/flutter_tools/gradle/flutter.gradle]: Done!

**Comment by cbracken** [comment]
Fixes #18278 (overall tracking bug #17051).

**Review by tvolkert — APPROVED** [review]
We should also be able to get rid of the *_dart1 benchmarks in the devicelab manifest.yaml.

**Review by tvolkert — APPROVED** [review]
LGTM

After this, we get to delete the `new` usages!

**Pull request merged**